extensions/xy_grid/script.py

```diff
@@ -146,274 +148,131 @@ def parse_axis(axis, value):
+            if axis_type['x'] == "prompts":
+                for new in chatbot_wrapper(j.strip().strip('"'), custom_state):
+                    gen_output = new
+            elif axis_type['y'] == "prompts":
+                for new in chatbot_wrapper(i.strip().strip('"'), custom_state):
+                    gen_output = new
+            else:
                 for new in chatbot_wrapper(custom_state['textbox'].strip(), custom_state):
                     gen_output = new
 
-                if len(gen_output) == 0:
-                    gen_output = [['','']]
-                user_output = convert_to_markdown(gen_output[-1][0])
-                bot_output = convert_to_markdown(gen_output[-1][1])
-
-                if custom_state['mode'] == 'instruct':
-                    output = output + f"<tr><th>{i.strip()}</th><td><b>{custom_state['name1']}</b> {user_output}<b>{custom_state['name2']}</b> {bot_output}</td></tr>"
-                else:
-                    output = output + f"<tr><th>{i.strip()}</th><td><h3><b>{custom_state['name1']}:</b></h3> {user_output}<h3><b>{custom_state['name2']}:</b></h3> {bot_output}</td></tr>"
-
-                # Remove the last outputs, so they don't influence future generations
-                gen_output.pop()
+            # Sometimes it the generation kicks back nothing and it causes problems
+            if len(gen_output) == 0:
+                gen_output = [['','']]
+
+            # Turn the output into HTML for our table
+            user_output = convert_to_markdown(gen_output[-1][0])
+            bot_output = convert_to_markdown(gen_output[-1][1])
+            if custom_state['mode'] == 'instruct':
+                output = output + f"<td><h3><b>{custom_state['name1']}</b></h3> {user_output}<h3><b>{custom_state['name2']}</b></h3> {bot_output}</td>"
+            else:
+                output = output + f"<td><h3><b>{custom_state['name1']}:</b></h3> {user_output}<h3><b>{custom_state['name2']}:</b></h3> {bot_output}</td>"
+
+            # Remove the last outputs, so they don't influence future generations
+            if custom_state['mode'] == 'instruct':
+                shared.history['internal'].clear()
+                shared.history['visible'].clear()
+            else:
                 if len(shared.history['internal']) > 1:
                     shared.history['internal'].pop()
-
-        else:
-            return "<h1><span style=\"color: red;\">ERROR: both fields are empty</span>"
-
+                elif len(shared.history['internal']) == 1:
+                    if shared.history['internal'] == gen_output:
+                        shared.history['internal'].clear()
+                if len(shared.history['visible']) > 1:
+                    shared.history['visible'].pop()
+                elif len(shared.history['visible']) == 1:
+                    if shared.history['visible'] == gen_output:
+                        shared.history['visible'].clear()
```

Comment:
@kaiokendev I fixed the context poisoning with this block here. Can you test the new script and see if you find any bugs?

Reply:
I'm not sure how or if I can show more lines of code with my comment, but the instruct portion is at the top of this block, a few lines up.

Reply:
Works correctly! one thing I forgot to mention before -- in the output file saving you should change from
>mkdir

to 
>os.makedirs(output_folder)

So it can make the intermediate folders if they don't already exists
Thanks for fixing this

Reply:
Alright, I'll do that. I was considering putting this in an `outputs/xy_grid` folder in the root directory, but I don't know how @oobabooga would feel about that. Personally, I hate when extensions output things to their own folder in SD, but it felt a little presumptuous.

I also would like to include models as an axis option, but it's gonna take a little work to figure out exactly how I want to go about getting the correct instruct characters lined up with the correct models. If you were to load three different models with 3 different instruct characters, it would be pretty much useless for 2/3 of the generations.